server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -540,13 +559,81 @@ private static SnapshotsInRepo sortAndFilterSnapshots(
             final Predicate<SnapshotInfo> fromSortValuePredicate = buildFromSortValuePredicate(sortBy, fromSortValue, order, null, null);
             if (predicate == null) {
                 predicate = fromSortValuePredicate;
-            } else {
+            } else if (fromSortValuePredicate != null) {
                 predicate = fromSortValuePredicate.and(predicate);
             }
         }
         return predicate;
     }
 
+    // Builds a predicate that can be applied to a combination of snapshot id and repository data to filter out snapshots that are not
+    // required to answer the request before loading their SnapshotInfo from the repository
+    // TODO: extend this method to cover the pagination after value as well where possible
+    @Nullable
+    private static BiPredicate<SnapshotId, RepositoryData> buildSnapshotPreflightPredicate(
+        GetSnapshotsRequest.SortBy sortBy,
+        SortOrder order,
+        String fromSortValue
+    ) {
+        if (fromSortValue == null) {
+            return null;
+        }
+        switch (sortBy) {
+            case START_TIME:
+                final long after = Long.parseLong(fromSortValue);
+                return order == SortOrder.ASC ? (snapshotId, repositoryData) -> {
+                    final RepositoryData.SnapshotDetails details = repositoryData.getSnapshotDetails(snapshotId);
+                    if (details == null) {
```

Comment:
Do we have tests generating data with no details as well as with details but `-1` timestamp?

Reply:
We don't unfortunately. I'll try adding some. We should have the infrastructure to do it with reasonable effort.

Reply:
> We should have the infrastructure to do it with reasonable effort

Turns out that we do not. We have infrastructure for testing this, but it's broken in a subtle way causing it to not actually test missing information here (it's build using the test-repo `BlobStoreRepositoryTest` but doesn't work because the repo never writes `RepositoryData` without the details at this point). 
Adding tests for this case requires fixing that functionality which is a bigger piece of work I think. Given how the fact that the repository regenerates the snapshot details so efficiently these days it seems like quite the edge case and may be ok to test in a follow-up? :)